### PR TITLE
Vendor Merlin

### PR DESCRIPTION
**Pull request by lthms**
This patch removes the codebase of Merlin from the repository, and add
it back as a git submodule.  We use the fork used by `ocaml-lsp`, that
has the particularity to assign a public name to Merlin libraries.
Other than that, Merlin is left unchanged.  We need to adapt
coq-of-ocaml to that.  It turns out that the `assert false` reached by
coq-of-ocaml can be avoided by using `Mpipeline.with_pipeline`.

### CODE REVIEW

File: src/mixedPath.ml [hunk 1]
```diff
@@ -1,6 +1,15 @@
 (** A [PathName.t], eventually followed by accesses inside first-class modules. *)
 open SmartPrint
 open Monad.Notations
+open Path
```
Comment (clarus) [inline src/mixedPath.ml:4]: Without opening `Path`, for uniformity.

**Comment by clarus** [comment]
Thanks very much for this PR, and for finding how to avoid the `assert`! I am not a big fan of sub-modules in general, but this seems like a good solution to avoid copy-pasting hundreds of files and staying up-to-date with the upstream Merlin. Plus `ocaml-lsp` seems to be here to stay. :raised_hands:

**Comment by lthms** [comment]
I’ve updated the patch as follows

```patch
diff --git a/src/mixedPath.ml b/src/mixedPath.ml
index 31f0320..fdc2892 100644
--- a/src/mixedPath.ml
+++ b/src/mixedPath.ml
@@ -1,9 +1,9 @@
 (** A [PathName.t], eventually followed by accesses inside first-class modules. *)
 open SmartPrint
 open Monad.Notations
-open Path
 
 let path_to_string_list p =
+  let open Path in
   let rec aux acc = function
     | Pident id -> Ident.name id :: acc
     | Pdot (p, str) -> aux (str :: acc) p
```

Is it better?

**Comment by clarus** [comment]
OK yes! Waiting for the CI to go through.

**Pull request merged**